aptos-move/block-executor/src/executor.rs

```diff
@@ -1662,56 +1546,178 @@ where
 
     /// Common finalization logic for both BlockSTM and BlockSTMv2 parallel execution.
     /// Handles commit task validation, error checking, state updates, and cleanup.
+    /// maybe_executor must be initialized if there was no error during parallel execution.
+    #[allow(clippy::too_many_arguments)]
     fn finalize_parallel_execution(
         &self,
+        maybe_executor: Option<E>,
+        signature_verified_block: &TP,
         shared_maybe_error: &AtomicBool,
         has_remaining_commit_tasks: bool,
         final_results: ExplicitSyncWrapper<Vec<E::Output>>,
-        block_epilogue_txn: Option<T>,
+        block_limit_processor: ExplicitSyncWrapper<BlockGasLimitProcessor<T, S>>,
+        block_epilogue_txn_idx: Option<TxnIndex>,
```

Comment:
I'm a little bit confused on this index here.

Let's say a block has 4 txns as input (0, 1, 2, 3), 1 is SkipRest (due to epoch change or gas limit), so 2 and 3 will be retry.

What will the epilogue index be after your change?

Reply:
tagged you, it happens in materialize_txn_idx.